provisioning/WFLY-21122-WildFly-Feature_pack-Family.adoc

```diff
@@ -0,0 +1,214 @@
+
+Each feature-pack would expose the following set of criteria:
+
+* WildFly EE feature-pack: `jakarta-ee-deployment`, `jakarta-ee`, `jakarta-ee-10`
+* WildFly feature-pack: `microprofile`, `microprofile-7.1` + the following ones inherited from WildFly EE : `jakarta-ee-deployment`, `jakarta-ee`
+* WildFly Preview feature-pack: `jakarta-ee-deployment`, `jakarta-ee`, `jakarta-ee-11`, `microprofile`, `microprofile-7.1`
+
+=== Hard Requirements
+
+* Allow to provision WildFly or WildFly Preview with other feature-packs compatible with the defined criteria.
+* Suppress (or at least deprecate and remove in a next version) the extra Preview feature-packs that have been defined to work with WildFly Preview: 
+cloud Preview, datasources Preview and GRPC Preview.
+* No enforcement for extra feature-packs to benefit from this feature. Already released feature-pack could be provisioned with new WildFly feature-packs
+
+NOTE: That is outside of this proposal but WildFly and WildFly Preview feature-packs, 
+in order to be valid alternatives, must expose a common set of Galleon `features`, `packages` and `layers`.
+
+=== Implementation notes
+
+The changes in the WildFly project are:
+
+* Bump to Galleon 7.x and WildFly Galleon Plugins 8.1.x
+* Update the three `wildfly-feature-pack-build.xml` files to define their family and criteria.
+* Update the WildFly full `wildfly-feature-pack-build.xml` file to inherit criteria from its WildFly EE dependency.
+
+`wildfly-feature-pack-build.xml` extract:
+
+```
+<build xmlns="urn:wildfly:feature-pack-build:3.5" producer="wildfly@maven(org.jboss.universe:community-universe):current">
+    <family name="wildfly">
+      <criteria name="microprofile"/>
+      <criteria name="microprofile-7.1"/>
+    </family>
+    <dependencies>
+        <dependency group-id="org.wildfly" artifact-id="wildfly-ee-galleon-pack" allowedFamily="wildfly:jakarta-ee-deployment+jakarta-ee">
+    ...
+</build>
+```
+
+The `allowedFamily` attribute defines the criteria that are inherited from `wildfly-ee-galleon-pack`. We can see that `jakarta-ee-11` is not inherited.
+This is a choice to not bound WildFly feature-pack to a specific version of Jakarta EE. This choice could be revisited.
+
+An example of a feature-pack that depends on WildFly EE 10:
+
+```
+<build xmlns="urn:wildfly:feature-pack-build:3.5" producer="org.example:myfeaturepack">
+    <dependencies>
+        <dependency group-id="org.wildfly" artifact-id="wildfly-ee-galleon-pack" allowedFamily="wildfly:jakarta-ee-10">
+    <dependencies>
+    ...
+</build>
+```
+
+==== Potential impact for WildFly Glow
+
+Today WildFly Glow CLI and its integration in the WildFly Maven Plugin offer the `wildfly-preview` option to contextualize the feature-packs discovery to use WildFly Preview feature-packs. This option should be deprecated
+and a new option `wildfly-variant=<variant name>` should be introduced. An immediate variant is `preview`, but new ones could be introduced in the future.
```

Comment:
Do we need to do this as part of this RFE? It seems like a separate thing.

I think what you're thinking about is when we have 2 variants of std WF and 3 total, then you need a key/value type param. Which makes sense.

But I think when that happens this setting becomes a way of saying "variant of FP that provides jakarata-ee-##+" where "##" is whatever the current baseline is. But what the best 'key' is to describe that is not obvious, and what the values will be when we have those 3 variant is not obvious. So maybe think about it more?